From 980306471fee903863c399741a38aa898c335979 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Sat, 17 Oct 2020 17:56:33 -0700
Subject: [PATCH] Improve TZUpdater and Python links

* tz-link.html: Improve URLs to TZUpdater (thanks to
Brian Inglis) and modernize the discussion of TZUpdater
and some of its competitors.  Likewise for Python.
---
 tz-link.html | 35 ++++++++++++++++++++---------------
 1 file changed, 20 insertions(+), 15 deletions(-)

diff --git a/tz-link.html b/tz-link.html
index 606a83b..a0e35a0 100644
--- a/tz-link.html
+++ b/tz-link.html
@@ -245,13 +245,12 @@ href="https://play.google.com/store/apps/details?id=com.google.android.timezone.
 <a
 href="http://userguide.icu-project.org/datetime/timezone"><abbr
 title="International Components for Unicode">ICU</abbr></a>,
-<a href="https://developer.ibm.com/javasdk/support/dst/jtzu/"><abbr>IBM</abbr></a>
-and <a
-href="https://www.oracle.com/technetwork/java/javase/tzupdater-readme-136440.html">Oracle</a>
-Java, <a href="https://www.joda.org/joda-time/tz_update.html">Joda-Time</a>, <a
+<a href="https://developer.ibm.com/javasdk/support/dst/jtzu/"><abbr>IBM</abbr>
+JDK</a>,
+<a href="https://www.joda.org/joda-time/tz_update.html">Joda-Time</a>, <a
 href="https://dev.mysql.com/doc/refman/en/time-zone-support.html">MySQL</a>,
-and <a
-href="https://nodatime.org/userguide/tzdb">Noda Time</a> (see below).
+<a href="https://nodatime.org/userguide/tzdb">Noda Time</a>, and <a
+href="https://www.oracle.com/java/technologies/javase/tzupdater-readme.html#usage">OpenJDK/Oracle JDK</a>.
 </p>
 <p>Sources for the <code><abbr>tz</abbr></code> database are
 <a href="https://en.wikipedia.org/wiki/UTF-8"><abbr
@@ -436,9 +435,13 @@ freely available under the <abbr>MIT</abbr> license.</li>
 <li>Java-based compilers and libraries include:
 <ul>
 <li>The <a
-href="https://www.oracle.com/technetwork/java/javase/tzupdater-readme-136440.html">TZUpdater
+href="https://www.oracle.com/java/technologies/javase/tzupdater-readme.html">TZUpdater
 tool</a> compiles <code><abbr>tz</abbr></code> source into the format used by
-Oracle Java.</li>
+<a href="https://openjdk.java.net/">OpenJDK</a> and
+<a href="https://jdk.java.net/">Oracle JDK</a>.
+Although its source code is proprietary, its executable is available under the
+<a href="https://www.oracle.com/a/tech/docs/tzupdater-lic.html">Java SE
+Timezone Updater License Agreement</a>.</li>
 <li>The <a
 href="https://www.oracle.com/technetwork/articles/java/jf14-date-time-2125367.html">Java
 SE 8 Date and Time</a> <abbr>API</abbr> can be supplemented by <a
@@ -451,11 +454,10 @@ and time <abbr>API</abbr></a> contains a class
 Java 8 <code>java.time</code>, which its users should migrate to once
 they can assume Java 8 or later. It is available under the <a
 href="https://www.apache.org/licenses/LICENSE-2.0">Apache License</a>.</li>
-<li><a href="https://bell-sw.com/pages/iana-updater/">IANA Updater</a>,
-<a href="https://github.com/akashche/tzdbgen">tzdbgen</a>, and <a
+<li><a href="https://bell-sw.com/pages/iana-updater/">IANA Updater</a> and <a
 href="https://www.azul.com/products/open-source-tools/ziupdater-time-zone-tool/">ZIUpdater</a>
-are other alternatives to TZUpdater. IANA Updater's license is unclear;
-the others are licensed under the <abbr>GPL</abbr>.</li>
+are alternatives to TZUpdater. IANA Updater's license is unclear;
+ZIUpdater is licensed under the <abbr>GPL</abbr>.</li>
 <li><a href="https://github.com/MenoData/Time4A">Time4A: Advanced date and
 time library for Android</a> and
 <a href="https://github.com/MenoData/Time4J/">Time4J: Advanced date,
@@ -515,7 +517,10 @@ It is freely available under a <abbr>BSD</abbr>-style license.</li>
 <li><a href="http://pytz.sourceforge.net">pytz &ndash; World Timezone
 Definitions for Python</a> compiles <code><abbr>tz</abbr></code> source into
 <a href="https://www.python.org">Python</a>.
-It is freely available under a <abbr>BSD</abbr>-style license.</li>
+It is freely available under a <abbr>BSD</abbr>-style license.
+In code that can assume Python 3.9 or later it is superseded by
+<a href="#python-zoneinfo"><code>zoneinfo.ZoneInfo</code></a>
+and the <a href="#pypi-tzdata"><code>tzdata</code> Python package</a>.</li>
 <li><a href="https://tzinfo.github.io">TZInfo &ndash;
 Ruby Timezone Library</a>
 compiles <code><abbr>tz</abbr></code> source into
@@ -579,14 +584,14 @@ DateTime::TimeZone::Tzfile</a>
 is a <abbr>TZif</abbr> file reader written in Perl.
 It is freely available under the same terms as Perl
 (dual <abbr>GPL</abbr> and Artistic license).</li>
-<li>Python has a <a
+<li>Python has a <a id="python-zoneinfo"
 href="https://docs.python.org/3.9/library/zoneinfo.html"><code>zoneinfo.ZoneInfo</code>
 class</a> that reads <abbr>TZif</abbr> data and creates objects
 that represent <code><abbr>tzdb</abbr></code> timezones.
 Python is freely available under the
 <a href="https://docs.python.org/3.9/license.html">Python Software Foundation
 License</a>.
-A companion <a href="https://pypi.org/">PyPI</a> module
+A companion <a id="pypi-tzdata"> href="https://pypi.org/">PyPI</a> module
 <a href="https://pypi.org/project/tzdata/"><code>tzdata</code></a>
 supplies TZif data if the underlying system data cannot be found;
 it is freely available under the Apache License.</li>
-- 
1.8.3.1

